### PR TITLE
Friends endpoint/services

### PR DESCRIPTION
**Pull request by lseiller**

### CODE REVIEW

**Comment by lseiller** [comment]
This PR will close #2

File: api/src/friends/friends.controller.ts [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+      }
+    },
+    isArray: true
+  })
+  @ApiNotFoundResponse({
+    description: "No request found."
+  })
+  @UseGuards(JwtAuthGuard)
+  @Get('request')
+  async findAllRequest(@Request() req): Promise<Selectable<FriendRequest>[]> {
```
Comment (Standlc) [inline api/src/friends/friends.controller.ts:76]: The front will need more infos about the friends requests (username, avatar, etc)

File: api/src/friends/friends.controller.ts [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+          type: 'integer'
+        }
+      }
+    },
+    isArray: true
+  })
+  @ApiNotFoundResponse({
+    description: "No friend found."
+  })
+  @UseGuards(JwtAuthGuard)
```
Comment (Standlc) [inline api/src/friends/friends.controller.ts:153]: Not necessarily protected endpoint (users may want to see other user's friends)

**Review by Standlc — APPROVED** [review]
Ok, left some comments!

**Pull request merged**